ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactionsTest.java

```diff
@@ -276,6 +276,25 @@ public void selectTransactionsUntilPendingIsEmpty() {
     assertThat(parsedTransactions.get(1)).isEqualTo(transaction1);
   }
 
+  @Test
+  public void shouldNotSelectReplacedTransactionWithSameSenderAndNonceButGreaterGas() {
+    final Transaction transaction1 = transactionWithNonceSenderAndGasPrice(1, KEYS1, 1);
+    final Transaction transaction2 = transactionWithNonceSenderAndGasPrice(1, KEYS1, 2);
+
+    transactions.addRemoteTransaction(transaction1);
+    transactions.addRemoteTransaction(transaction2);
+
+    final List<Transaction> parsedTransactions = Lists.newArrayList();
+    transactions.selectTransactions(
+        transaction -> {
+          parsedTransactions.add(transaction);
+          return PendingTransactions.TransactionSelectionResult.CONTINUE;
+        });
+
+    assertThat(parsedTransactions.size()).isEqualTo(1);
```

Comment:
can you do a assertThat(parsedTransactions).containsExactly(transaction2)

Reply:
that's nicer.
done